sycl/source/backend.cpp

```diff
@@ -72,8 +72,16 @@ platform make_platform(pi_native_handle NativeHandle, backend Backend) {
 
   // Create PI platform first.
   pi::PiPlatform PiPlatform = nullptr;
-  Plugin->call<PiApiKind::piextPlatformCreateWithNativeHandle>(NativeHandle,
-                                                               &PiPlatform);
+  sycl::detail::pi::PiResult Result =
+      Plugin->call_nocheck<PiApiKind::piextPlatformCreateWithNativeHandle>(
+          NativeHandle, &PiPlatform);
+  if (Result == PI_ERROR_INVALID_OPERATION) {
+    throw sycl::exception(
+        sycl::make_error_code(sycl::errc::feature_not_supported),
+        "Platform create with native handle command not supported by backend.");
+  } else {
+    Plugin->checkPiResult(Result);
+  }
```

Comment:
This pattern seems to repeat a lot. Could we maybe make a new `call_optional` or `call_ext` or something along those lines?

Reply:
Thanks for your suggestion, it certainly would be good to refactor out the repetitiveness here. I wonder if it might be worth putting the check inside the pre-existing `call` or `call_nocheck` functions?